ledger/ledger-api-tests/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_8/ConfigManagementServiceIT.scala

```diff
@@ -135,60 +135,36 @@ final class ConfigManagementServiceIT extends LedgerTestSuite {
   )(implicit ec => { case Participants(Participant(ledger)) =>
     for {
       // Get the current time model
-      response1 <- ledger.getTimeModel()
+      preUpdateTimeModelResponse <- ledger.getTimeModel()
       oldTimeModel = {
-        assert(response1.timeModel.isDefined, "Expected time model to be defined")
-        response1.timeModel.get
+        assert(preUpdateTimeModelResponse.timeModel.isDefined, "Expected time model to be defined")
+        preUpdateTimeModelResponse.timeModel.get
       }
 
       // Set a new time model with the next generation in parallel
       t1 <- ledger.time()
       f1 = ledger.setTimeModel(
         mrt = t1.plusSeconds(30),
-        generation = response1.configurationGeneration,
+        generation = preUpdateTimeModelResponse.configurationGeneration,
         newTimeModel = oldTimeModel,
       )
       f2 = ledger.setTimeModel(
         mrt = t1.plusSeconds(30),
-        generation = response1.configurationGeneration,
+        generation = preUpdateTimeModelResponse.configurationGeneration,
         newTimeModel = oldTimeModel,
       )
 
-      failure <- f1
+      _ <- f1
         .flatMap(_ => f2)
         .mustFail("setting Time Model with an outdated generation")
 
       // Check if generation got updated (meaning, one of the above succeeded)
-      response2 <- ledger.getTimeModel()
+      postUpdateTimeModelResponse <- ledger.getTimeModel()
     } yield {
       assert(
-        response1.configurationGeneration + 1 == response2.configurationGeneration,
-        s"New configuration's generation (${response2.configurationGeneration} should be original configurations's generation (${response1.configurationGeneration} + 1) )",
+        preUpdateTimeModelResponse.configurationGeneration + 1 == postUpdateTimeModelResponse.configurationGeneration,
+        s"New configuration's generation (${postUpdateTimeModelResponse.configurationGeneration} should be original configurations's generation (${preUpdateTimeModelResponse.configurationGeneration} + 1) )",
       )
-      Try {
-        // if the "looser" command fails already on command submission (the winner completed before looser submission is over)
-        assertGrpcError(
-          failure,
-          LedgerApiErrors.RequestValidation.InvalidArgument,
-          Some("Mismatching configuration generation"),
-        )
-      }.recover { case _ =>
-        // if the "looser" command fails after command submission (the winner completed after looser did submit the configuration change)
-        assertGrpcError(
-          failure,
-          LedgerApiErrors.Admin.ConfigurationEntryRejected,
-          Some("Generation mismatch"),
-        )
```

Comment:
It's not flaky, but I find that the cost of extending and maintaining this test is more than the value it provides.s

The main change is that on our side this will now return another self-service error as well, and getting to the point where we assert 3 possible return values from a call is definitely a sign of un-maintainable tests.

This leads to the other issue where we misuse conformance tests as integration tests, with the main reasoning behind removing the assertions being the following:

- From a ledger conformance perspective I would expect that we validate that no two concurrent configurations are accepted with the same generation
- Self-service error codes are a participant only feature and we should have integration/contract tests that validate the participant output, in a more controlled environment where we can set the expectaction for a single status code instead of defaulting through multiple possible outputs
- Adding the extract assertion for a possible post-execution conflicts self-service error code to this test case feels like diluting the scope of the test a lot, having assertions that are unrelated with the scope of the test (while the error is triggered in this case, it's only because of conccurrent submission and has no relation to config managemnt itself)

I fully aggree that not testing self service error codes is not something that we should strive for, but at the same time we can't spread out the testing for this feature all throughtout the conformance tests, but it should be done in a more structured mode. 

Let me know what are your thoughts on how we would proceed in the future with this type of tests, I'm also more than happy just to add a new possible error code (will at least refactor to assert only error code and not the description message)

Reply:
Self-service error codes are meant to help the app/client or system operator understand what happened when a request went bad. Aside from a couple of errors that are outside the user's control, most errors are related to a request's structure or timing. Ideally, the error code id should generally identify the problem cause and have additional information (e.g. resource, error message) specialize the error to a specific component/mechanism. So, I wouldn't say that self-service error codes are a participant-only feature. 

Looking at this test: it seems we have two possible error codes for the same error condition. Ideally, we should merge them (i.e. only have a `CONFIGURATION_GENERATION_MISMATCH`). I will look into that. 

Happy to sync further on this. 

P.S. Is the error code that you need to introduce describing a new error condition or can we reuse and existing one?

Reply:
Good point that the error codes are not participant only. That makes sense. 

The error code is a new one that is triggered by any post-execution conflicts.